backend/infrahub/core/schema/attribute_parameters.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+from pydantic import Field
+
+from infrahub.core.constants.schema import UpdateSupport
+from infrahub.core.models import HashableModel
+
+
+def get_attribute_parameters_class_for_kind(kind: str) -> type[AttributeParameters]:
+    return {
+        "Text": TextAttributeParameters,
+        "TextArea": TextAttributeParameters,
+    }.get(kind, AttributeParameters)
+
+
+class AttributeParameters(HashableModel):
+    class Config:
+        extra = "forbid"
+
+
+class TextAttributeParameters(AttributeParameters):
+    regex: str | None = Field(
+        default=None,
+        description="Regular expression that attribute value must match if defined",
+        json_schema_extra={"update": UpdateSupport.VALIDATE_CONSTRAINT.value},
+    )
+    min_length: int | None = Field(
+        default=None,
+        description="Set a minimum number of characters allowed.",
+        json_schema_extra={"update": UpdateSupport.VALIDATE_CONSTRAINT.value},
+    )
+    max_length: int | None = Field(
```

Comment:
I think that we might actually want separate TextAttributeParameters and TextAreaAttributeParameters due to the max length in order to set different upper limits in the schema. I.e. a user shouldn't really be able to set a max_length for a Text kind that's longer than our own upper limit with regards to the index restriction. This could be something that we change in an upcoming PR though as not setting any limit would basically mean the same thing.